drivers/mfd/mfd_max31790.c

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2024 SILA Embedded Solutions GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT maxim_max31790
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/mfd/max31790.h>
+#include <zephyr/drivers/pwm.h>
+#include <zephyr/drivers/pwm/max31790.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(max_max31790, CONFIG_MFD_LOG_LEVEL);
+
+struct max31790_config {
+	struct i2c_dt_spec i2c;
+};
+
+struct max31790_data {
+};
+
+static void max31790_set_globalconfiguration_i2cwatchdog(uint8_t *destination, uint8_t value)
+{
+	uint8_t length = MAX37190_GLOBALCONFIGURATION_I2CWATCHDOG_LENGTH;
+	uint8_t pos = MAX37190_GLOBALCONFIGURATION_I2CWATCHDOG_POS;
+
+	*destination &= ~GENMASK(pos + length - 1, pos);
+	*destination |= FIELD_PREP(GENMASK(pos + length - 1, pos), value);
+}
+
+int max31790_read_register_uint8(const struct device *dev, uint8_t address, uint8_t *value)
+{
+	const struct max31790_config *config = dev->config;
+	int result;
+
+	result = i2c_reg_read_byte_dt(&config->i2c, address, value);
+	if (result != 0) {
+		LOG_ERR("unable to read register 0x%02X, error %i", address, result);
+	}
+
+	LOG_DBG("read value 0x%02X from register 0x%02X", *value, address);
+
+	return result;
+}
+
+int max31790_read_register_uint16(const struct device *dev, uint8_t address, uint16_t *value)
+{
+	const struct max31790_config *config = dev->config;
+	int result;
+
+	result = i2c_write_read_dt(&config->i2c, &address, sizeof(address), value, sizeof(*value));
+	if (result != 0) {
+		LOG_ERR("unable to read from address 0x%02X, error %i", address, result);
+	}
+
+	*value = sys_be16_to_cpu(*value);
+
+	LOG_DBG("read value 0x%04X from address 0x%02X", *value, address);
+
+	return result;
+}
+
+int max31790_write_register_uint8(const struct device *dev, uint8_t address, uint8_t value)
```

Comment:
I think this would be a good opportunity to tidy up some of the interfaces in this driver.
This wrapper in unnecessary, the caller should use i2c_reg_write_byte_dt directly.

Reply:
I actually need these wrappers, because I'm calling these functions from the sub devices (PWM and sensor). Otherwise I would have to make `struct max31790_config` public.

Reply:
For simple cases you can keep the i2c reference and use that directly, you don't need to go via the MFD parent.
e.g.: 
https://github.com/zephyrproject-rtos/zephyr/blob/a738bfa7f42219487c5051eeae2a59de5ab9e283/drivers/regulator/regulator_npm6001.c#L74

Reply:
done